src/Service/SF1500/BrugerService.php

```diff
@@ -49,80 +56,153 @@ public function list(array $ids, array $fields = []): array
+            ->setFoersteResultatReference((int)($query['offset'] ?? 0))
             ->setAttributListe($attributListe)
             ->setRelationListe($relationListe);
 
-        return $this->clientSoeg()->soeg($request);
+        $response = $this->clientSoeg()->soeg($request) ?: null;
+
+        if (null !== $response && isset($query[self::FILTER_LEDER])) {
+            // Keep or throw away managers.
+            $managerIds = $this->getManagerIds();
+
+            $brugerIds = $response->getIdListe()->getUUIDIdentifikator();
+            $response->setIdListe(new IdListeType(
+                array_values(
+                    $query[self::FILTER_LEDER]
+                        // Keep only managers.
+                        ? array_intersect($brugerIds, $managerIds)
+                        // Throw away managers.
+                        : array_diff($brugerIds, $managerIds)
+                )
+            ));
+        }
+
+        return $response;
+    }
+
+    private function getManagerIds()
+    {
+        $service = $this->getService(OrganisationFunktionService::class);
+        /** @var OrganisationFunktion[] $organisationFunktioner */
+        $organisationFunktioner = $service->soeg([
+            OrganisationFunktionService::FILTER_FUNKTIONSTYPEID => $this->options['organisation-funktion-manager-id']
+        ]);
+
+        return array_values(array_filter(array_map(
+            static fn (OrganisationFunktion $organisationFunktion) => $organisationFunktion->tilknyttedeBrugere[0] ?? null,
+            $organisationFunktioner
+        )));
     }
 
-    protected function doList(array $ids): ListOutputType
+    /**
+     * {@inheritdoc}
+     */
+    protected function doList(array $ids): ?ListOutputType
     {
         return $this->clientList()
-            ->_list_1((new ListInputType())
-                ->setUUIDIdentifikator($ids));
+            ->_list_1(new ListInputType($ids)) ?: null;
     }
 
-    protected function doLaes(string $id): LaesOutputType
+    /**
+     * {@inheritdoc}
+     */
+    protected function doLaes(string $id): ?LaesOutputType
     {
         return $this->clientLaes()
-            ->laes((new LaesInputType())
-                ->setUUIDIdentifikator($id));
+            ->laes(new LaesInputType($id)) ?: null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function buildModel($oejebliksbillede): Bruger
+    {
+        assert($oejebliksbillede instanceof FiltreretOejebliksbilledeType);
+        $id = $oejebliksbillede->getObjektType()->getUUIDIdentifikator();
+        $model = new Bruger(['id' => $id]);
+        foreach ($oejebliksbillede->getRegistrering() as $registrering) {
+            foreach ($registrering->getAttributListe()->getEgenskab() as $egenskab) {
+                $model->brugernavn = $egenskab->getBrugerNavn();
+                $model->brugertype = $egenskab->getBrugerTypeTekst();
+            }
```

Comment:
Do we unintentionally overwrite `brugernavn` and `brugertype` here?

Reply:
Not unintentionally, but we assume that we have only a single item in the attribute list. I'll add a clarifying comment.